### PR TITLE
Use environment file for initial conda env creation.

### LINKED ISSUE #1901: conda env creation fails if Python 3 selected
I put this in `readthedocs.yml`:

```yaml
conda:
    file: docs/conda-env.yml
python:
    version: 3
    setup_py_install: true
```

It creates a conda env with Python 3.5, and then tries to install the standard docs machinery. But docutils is pinned to 0.11, and there isn't a build of this for Python 3.5 (there is a package of docutils 0.12). So I see this failure:

```
conda install --yes --name docs-build-w-conda sphinx==1.3.1 Pygments==2.0.2 docutils==0.11 mock==1.0.1 pillow==3.0.0 sphinx_rtd_theme==0.1.7 alabaster>=0.7,<0.8,!=0.7.5
Fetching package metadata: /usr/local/lib/python2.7/dist-packages/requests/packages/urllib3/util/ssl_.py:315: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/usr/local/lib/python2.7/dist-packages/requests/packages/urllib3/util/ssl_.py:120: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
....
Solving package specifications: ..
Error: Unsatisfiable package specifications.
Generating hint: 
[                    ]|                                                  |   0%
[2/8                 ]|############                                      |  25%
[3/8                 ]|##################                                |  37%
[5/8                 ]|###############################                   |  62%
[6/8                 ]|#####################################             |  75%
[      COMPLETE      ]|##################################################| 100%



Hint: the following packages conflict with each other:
  - docutils ==0.11
  - python 3.5*

Use 'conda info docutils' etc. to see the dependencies for each package.
```

The obvious solution is not to pin docutils so it automatically picks the latest version available. If you prefer to keep it pinned, I think that installing dependencies at the same time as you create the environment should work; in this instance, it would fall back to Python 3.4 so it could satisfy the dependencies:

```
conda create --yes --name docs-build-w-conda python=3 sphinx==1.3.1 Pygments==2.0.2 docutils==0.11 ...
```

### PR DESCRIPTION
**Pull request by ericholscher**
Fixes #1901

### CODE REVIEW

File: readthedocs/doc_builder/python_environments.py [hunk 1]
```diff
@@ -174,23 +174,22 @@ def setup_base(self):
...
     def install_core_requirements(self):
-        conda_env_path = os.path.join(self.project.doc_path, 'conda')
 
         # Use conda for requirements it packages
         requirements = [
             'sphinx==1.3.1',
             'Pygments==2.0.2',
-            'docutils==0.11',
-            'mock==1.0.1',
+            'docutils',
```
Comment (jakirkham) [inline readthedocs/doc_builder/python_environments.py]: Couple of questions. Is it really ok to leave `docutils` unpinned or should it just be pinned to a newer version like `0.12`? Also, should `mock` really be dropped?

Comment (ericholscher) [inline readthedocs/doc_builder/python_environments.py]: The idea of including mock was to let folks mock dependencies, which presumably isn't an issue with conda.

We could also remove docutils entirely, and let Sphinx specify the docutils version. Not sure the best path forward.

Comment (jakirkham) [inline readthedocs/doc_builder/python_environments.py]: Ah, ok. So, you are not dependent on `mock`. In that case, I agree with you. Also, letting `sphinx` pull in `docutils` sounds find unless you are using it internally somehow.

Comment (ericholscher) [inline readthedocs/doc_builder/python_environments.py]: I've gone ahead and pinned it at 0.12 for now, as that's the least disruptive to how everything else works.

Comment (jakirkham) [inline readthedocs/doc_builder/python_environments.py]: Sorry to keep nitpicking on this, but if you are concerned about disruption then should the `mock` constraint simply be left as is?

**Comment by ericholscher** [comment]
Tests pass locally, think this is just Travis failing.

**Comment by agjohnson** [comment]
I've retriggered tests, if they are still failing I'll try to verify, but I'm assuming this is just travis flaking out as well

**Comment by jakirkham** [comment]
Another thing to consider, there is a `travis_wait` command. This will increase the wait time. ( https://docs.travis-ci.com/user/build-timeouts#Build-times-out-because-no-output-was-received )

**Pull request merged**